### ISSUE TITLE
First year shows dip even though 0 change / no data

### ISSUE THREAD

**Issue by cawarren** [Recidiviz/supervision-success-component#91]
**Describe the bug**

**To Reproduce**
1. Use the attached file for params.csv
2. Select New York, a state with no 2020 data
4. See chart - 2020 says '0 people' on hover, but the dot is at a noticeably lower place than the 2019 dot on the line.

**Expected behavior**
Since there is no 2020 data, I'd expect 2019 and 2020 to have the same population.

**Screenshots**
![image](https://user-images.githubusercontent.com/59526/127683043-fea8973c-1a3c-4ae7-b715-aca93af8c300.png)

**Desktop (please complete the following information):**
 - OS [e.g. Windows 10]: ChromeOS
 - Browser [e.g. chrome, edge]:  Chrome
 - Version [e.g. 22]: 91.0.4472.167 (Official Build) (64-bit)

**Additional context**

**Comment by cawarren** [comment]
It's interesting that it's anchored to that lower place even if the line goes up in 2021

![image](https://user-images.githubusercontent.com/59526/127683343-c4d53e80-74ce-4117-b39d-d72f06a00c9f.png)

Also, attaching actual params file now.
[params.csv](https://github.com/Recidiviz/supervision-success-component/files/6908888/params.csv)

**Comment by nasaownsky** [comment]
@cawarren can you please clarify the issue, what is wrong with '0 people'?
Currently, it is calculated as 'Total population' - 'baseline'.

**Comment by cawarren** [comment]
Nothing is wrong with '0 people', it's the shape of the line that seems wrong. If you look at the y-axis, it _looks like_ from 2019 to 2020 there is a decrease of ~10,000 people. But, we don't have data for 2020 (and the tooltip accurately says '0 people' / no change), so why does the line dip?

**Comment by nasaownsky** [comment]
@cawarren this is a decrease in 2019
01/01/2019: 46331
02/01/2019: 44973
03/01/2019: 43729
04/01/2019: 42589
05/01/2019: 41544
06/01/2019: 40586
07/01/2019: 39708
08/01/2019: 38903
09/01/2019: 38166
10/01/2019: 37490
11/01/2019: 36871
12/01/2019: 36303
01/01/2020: 35783

In https://github.com/Recidiviz/supervision-success-component/pull/77#pullrequestreview-705539822 we decided to center the 2020 data point.

**Comment by cawarren** [comment]
I think Josh's point in that PR was to center the dot in 2020 horizontally. I don't think we should try to center it vertically in the cross-hatch, that makes it look like the population went to a particular number on the y-axis when (in truth) we have no idea what it did.

@jessex let us know if this was intentional though and I'm missing something?

**Comment by jessex** [comment]
No my thought was _definitely_ to center the dot horizontally (which was still not the case at the time), but _not_ to center it vertically. I think that was a misunderstanding -- sorry about that! I didn't realize that it was centering vertically, and assumed (as noted in a comment thread) that it was something to do with the test data we had.

Let's get rid of the vertical centering!

**Comment by nasaownsky** [comment]
@cawarren @jessex the dot is not centered vertically, the screenshot above just has large minY and maxY values, so the Y-axis wouldn't rescale as implementation period changes. The dot is centered horizontally and the all data is right, issue is because this dot is at the BEGINNING of 2020 year (and this topic is covered in #77 PR).

**Comment by jessex** [comment]
I think I can see what you're saying, at least partially, @nasaownsky.

You are saying that because the `params.csv` file that Andrew provided includes no values for the `Revocations Population 2020...` and `Population Fraction 2020...` columns (F and G) for New York, the app concludes that there is no data for 2020, and so renders the cross-hatching treatment and such. But because that same file includes `Total Population 2020` and `New Offense 2020` values (columns M and O), then it does plug those numbers into the graph, leading to a lower data point for 2020? Can you confirm exactly which column in the one that the chart is using to render that 2020 data point for New York at a value of 35783?

Can you confirm that understanding, @nasaownsky, and share what that sequence of dates and values in your [previous comment](https://github.com/Recidiviz/supervision-success-component/issues/91#issuecomment-891229955) refers to? Is that what the projection logic is generating?

I think what may be happening here is potentially a mismapping of columns. If NY has the same `Total Population 2019` (L) and `Total Population 2020` (M) values, and has the same `New Offense 2019` (N) and `New Offense 2020` (O) values, then it doesn't seem right that the graph should decrease from 2019 to 2020. But instead, it kind of seems like what the app is doing is drawing a line from either L or M (both of which are 46331) to either N or O (both of which are 35783), which seems like incorrect logic.

CC @cawarren

**Comment by nasaownsky** [comment]
@jessex  the sequence of dates and values from the projection model https://docs.google.com/spreadsheets/d/1qa4Ts6Aw8MEpAN9vWLBeiUogRg_pmIKMnMxWKCcPOqA/edit#gid=1607410762
The first 12 months correspond to the 2019 calendar year.
![image](https://user-images.githubusercontent.com/20225344/128537637-9c5fb5c3-92f9-4153-a300-655f87078259.png)

**Comment by jessex** [comment]
@cawarren, what do you think here? You have a better understanding of the model and the product expectations than I do. Does this look right to you?

**Comment by cawarren** [comment]
Sorry, I'm still looking into this and am a bit befuddled by it - trying to have a response by end of day.

**Comment by cawarren** [comment]
Sorry for all of the confusion; this ended up being a mistaken formula in the spreadsheet from a change I made earlier. I've fixed the error, and the attached file is a true / correct / likely final params.csv. This issue no longer occurs with it, so we can close this issue.

![image](https://user-images.githubusercontent.com/59526/129292917-550b4c62-64a4-412e-a009-02b71ce3e40b.png)
[params.csv](https://github.com/Recidiviz/supervision-success-component/files/6979406/params.csv)
